### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by pcGarethJames**
Version 1 of the X-cart integration module

### CODE REVIEW

File: classes/XLite/Module/PureClarity/Personalization/View/Model/Admin/Dashboard/Signup.php [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                case 'timezone':
+                    $time = new DateTime('now', Converter::getTimeZone());
+                    $value = $time->getTimezone()->getName();
+                    if ($value == 'UTC') {
+                        $value = 'Europe/London';
+                    }
+                    break;
+                case 'url':
+                    $domain = ConfigParser::getOptions(['host_details', 'http_host']);
+                    $value = 'http://' . $domain;
```
Comment (PeterBrooksbank) [inline classes/XLite/Module/PureClarity/Personalization/View/Model/Admin/Dashboard/Signup.php]: Is this always going to return http for the store url?

File: classes/XLite/Module/PureClarity/Personalization/Core/Task/Signup.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+class Signup extends Periodic
+{
+    /**
+     * Returns the title of this task
+     *
+     * @return string
+     */
+    public function getTitle() : string
+    {
+        return static::t('PureClarity Sign up check');
```
Comment (PeterBrooksbank) [inline classes/XLite/Module/PureClarity/Personalization/Core/Task/Signup.php]: Why capital S but not for the other words?

**Review by PeterBrooksbank — CHANGES_REQUESTED** [review]
Few minor pieces

File: classes/XLite/Module/PureClarity/Personalization/View/Model/Admin/Dashboard/Signup.php (hunk 1 above)
Comment (pcGarethJames) [reply classes/XLite/Module/PureClarity/Personalization/View/Model/Admin/Dashboard/Signup.php]: yes, fixed now though :) it'll return https if the store has https enabled

File: classes/XLite/Module/PureClarity/Personalization/Core/Task/Signup.php (hunk 2 above)
Comment (pcGarethJames) [reply classes/XLite/Module/PureClarity/Personalization/Core/Task/Signup.php]: typo :P, fixed

**Review by PeterBrooksbank — APPROVED** [review]
LGTM

**Pull request merged**